core/primitives/src/transaction.rs

```diff
@@ -348,6 +348,20 @@ pub struct ExecutionOutcome {
     /// NOTE: Should be the latest field since it contains unparsable by light client
     /// ExecutionStatus::Failure
     pub status: ExecutionStatus,
+    /// Execution metadata, versioned
+    pub metadata: ExecutionMetadata,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Clone, Eq, Debug)]
+pub enum ExecutionMetadata {
+    // V1: Empty Metadata
+    ExecutionMetadataV1,
+}
+
+impl Default for ExecutionMetadata {
+    fn default() -> Self {
+        ExecutionMetadata::ExecutionMetadataV1
+    }
 }
```

Comment:
I think it's better to remove this default impl -- we want to include metadata with every outcome, so we'll have to explicitly create a metadata instance at every call-site anyway.

Reply:
I remembered the reason for this is ExecutionOutcome and structs containing ExecutionOutcome are deriving(Default), so ExecutionMetadata has to impl Default

Reply:
Yeah, than our hand is forced here! I wonder what is the root cause of ExecutionOutcome needing to be Default, but that's out of scope here!